xla/stream_executor/gpu/gpu_blas_lt.cc

```diff
@@ -0,0 +1,123 @@
+#if GOOGLE_CUDA
+#include "tsl/platform/tensor_float_32_utils.h"
+#endif
+
+namespace stream_executor::gpu {
+
+tsl::StatusOr<blas::DataType> AsBlasDataType(xla::PrimitiveType dtype) {
+  switch (dtype) {
+    case xla::PrimitiveType::F8E5M2:
+      return blas::DataType::kF8E5M2;
+    case xla::PrimitiveType::F8E4M3FN:
+      return blas::DataType::kF8E4M3FN;
+    case xla::PrimitiveType::S8:
+      return blas::DataType::kInt8;
+    case xla::PrimitiveType::F16:
+      return blas::DataType::kHalf;
+    case xla::PrimitiveType::BF16:
+      return blas::DataType::kBF16;
+    case xla::PrimitiveType::F32:
+      return blas::DataType::kFloat;
+    case xla::PrimitiveType::S32:
+      return blas::DataType::kInt32;
+    case xla::PrimitiveType::F64:
+      return blas::DataType::kDouble;
+    case xla::PrimitiveType::C64:
+      return blas::DataType::kComplexFloat;
+    case xla::PrimitiveType::C128:
+      return blas::DataType::kComplexDouble;
+    default:
+      return xla::InternalError("AsBlasDataType: unsupported type: %s",
+                           xla::primitive_util::LowercasePrimitiveTypeName(dtype));
+  }
+}
+
+tsl::StatusOr<blas::ComputationType> GetBlasComputationType(
+    xla::PrimitiveType lhs_dtype, xla::PrimitiveType output_dtype,
+    int64_t compute_precision) {
+  switch (output_dtype) {
+    case xla::PrimitiveType::F8E5M2:    // fall-through
+    case xla::PrimitiveType::F8E4M3FN:  // fall-through
+    case xla::PrimitiveType::F16:       // fall-through
+    case xla::PrimitiveType::BF16:
+      // Accumulate in f32 precision.
+      return blas::ComputationType::kF32;
+    case xla::PrimitiveType::F32:  // fall-through
+    case xla::PrimitiveType::C64:
+#if GOOGLE_CUDA
+      if (tsl::tensor_float_32_execution_enabled() && compute_precision <= 1 &&
+          lhs_dtype == output_dtype) {
+        // CublasLt requires compute type to be F32 for F8 matmul.
+        // TF32 should only be chosen for FP32 or C64 gemm
+        return blas::ComputationType::kTF32AsF32;
+      }
+#endif
+      return blas::ComputationType::kF32;
+    case xla::PrimitiveType::F64:  // fall-through
+    case xla::PrimitiveType::C128:
+      return blas::ComputationType::kF64;
+    case xla::PrimitiveType::S32:
+      return blas::ComputationType::kI32;
+    default:
+      return xla::InternalError("GetBlasComputationType: unsupported type");
+  }
+}
+
+// BLAS GeMM's output is column-major. If we require row-major, use identity:
+// C^T = (A @ B)^T = B^T @ A^T.
+bool MakeOutputColumnMajor(MatrixLayout& lhs, MatrixLayout& rhs,
+                           MatrixLayout& output, MatrixLayout *pC) {
+  bool swap_operands = output.order != MatrixLayout::Order::kColumnMajor;
+  if (swap_operands) {
+    std::swap(lhs, rhs);
+    rhs.Transpose();
+    lhs.Transpose();
+    // prevent pC and output from being swapped two times if they are equal!
+    if(pC != nullptr && pC != &output)
+      pC->Transpose();
```

Comment:
nit: please either use { } around this line, or move it to the previous line.
So 1-line if statements (with body) are ok, if it is a separate line, we prefer to use {} around the if body.